test/parallel_api/algorithm/alg.sorting/sort_by_key.pass.cpp

```diff
@@ -62,7 +99,55 @@ test_with_usm(sycl::queue& q)
     {
         if (i < (N - 1))
         {
-            EXPECT_TRUE(h_skey[i] >= h_skey[i + 1], "wrong sort result");
+            EXPECT_TRUE(h_skey[i] >= h_skey[i + 1], "wrong sort result with hetero policy, USM data");
+        }
+    }
+}
+
+void
+test_with_buffers(sycl::queue& q)
```

Comment:
can we generalize those functions? they are identical except memory allocation way and `begin/end` calls. I have at least two ideas how it can be done.

Anyway, all the comments from the top function are applicable to this one.